test/cloudtest/test_recovery.py

```diff
@@ -0,0 +1,252 @@
+    mz.suspend_k8s_node(node_names[0])
+
+
+@pytest.mark.long
+def test_unreplicated_storage_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated storage cluster is on the failed node. Queries of a downstream index in serializable mode should
+    continue to work but return stale data. Staleness should resolve within a minute or two.
+    """
+    cluster_setup = ClusterSetup(1)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.get_storage_name())
+
+    # expect (stale) data to be delivered timely
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index,
+        must_exceed_reached_index=False,
+        timeout_in_sec=30,
+    )
+
+    print("Expecting stalled data...")
+    # only request this index because the previous validation succeeded / did not block
+    stalled_index = get_current_counter_index(mz, cluster_setup)
+
+    print("Awaiting node recovery...")
+    # expect live data to be delivered at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        stalled_index,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+
+@pytest.mark.long
+def test_unreplicated_compute_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated compute cluster is on the failed node. Queries of indexes on the compute cluster should fail, but
+    resolve within a minute or two.
+    """
+    cluster_setup = ClusterSetup(2)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.get_compute_name())
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index + 1,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+
+@pytest.mark.long
+def test_replicated_compute_cluster_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    A replicated compute cluster is on the failed node. Queries of indexes on the compute cluster should experience no
+    disruption in latency, thanks to the second replica.
+    """
+    cluster_setup = ClusterSetup(3, compute_replication_factor=2)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.get_compute_name())
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index + 1,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
```

Comment:
This should be set to a very low value. As per the description of the scenario "queries should experience no disruption".

Reply:
Yes, you are right.

✅